administration/src/util/qrcode.ts

```diff
@@ -246,3 +248,20 @@ export const drawQRCode = (
     size
   )
 }
+
+export const convertProtobufToHexCode = (protobufQRCode: DynamicActivationCode): string => {
+  const qrCode = new QrCode({
+    qrCode: {
+      case: 'dynamicActivationCode',
+      value: protobufQRCode,
+    },
+  }).toBinary()
+  const qrCodeMatrix = encodeQRCode(qrCode).getMatrix()
+  return convertBitmapToHexmap(qrCodeMatrix)
+}
+
+const convertBitmapToHexmap = (matrix: QRCodeByteMatrix): string =>
+  matrix
+    .getArray()
+    .map(row => parseInt(row.join(''), 2).toString(16))
```

Comment:
See https://github.com/digitalfabrik/entitlementcard/issues/1321#issuecomment-1981174652

```suggestion
    .map(row => BigInt("0b" + row.join('')).toString(16))
```